tests/UnifiedSpecTests/UnifiedTestRunner.php

```diff
@@ -213,11 +213,10 @@ private function doTestCase(stdClass $test, string $schemaVersion, ?array $runOn
                 $operation = new Operation($o, $context);
                 $operation->assert();
             }
-
+        } finally {
             $context->stopEventObservers();
             $context->stopEventCollectors();
-        } finally {
-            $this->disableFailPoints($context->getFailPoints());
+            $context->disableFailPoints();
```

Comment:
Note that I moved `stopEventObservers()` and `stopEventCollectors()` _within_ the `finally` block. That seemed correct given the following text from [Executing a Test](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.md#executing-a-test) in the Unified Test Format spec:

> For each element in [test.operations](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.md#test_operations), follow the process in [Executing an Operation](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.md#executing-an-operation). If an unexpected error is encountered or an assertion fails, the test runner MUST consider this test to have failed.
>
> **If any event listeners were enabled on any client entities, the test runner MUST now disable those event listeners.**
>
> If any fail points were configured, the test runner MUST now disable those fail points (on the same server)...

It's _possible_ that this could have resulted in memory leaks after failing tests. The subscriber would never get unregistered, and since they were registered _globally_ they'd continue to accumulate in the process. Both EventCollector and EventObserver also hold a reference to a Context object, so those would also linger. I'm not sure if `gc_collect_cycles()` would have helped here, since it's PHPC itself that ultimately holds a reference to the subscriber(s) in its request-scoped HashTable.

I put this change in a separate commit, so if you think it deserves its own PHPLIB ticket for tracking feel free to create that and incorporate it into the message when merging this.